tools/spectral/ipa/rulesets/functions/getMethodReturnsSingleResource.js

```
@@ -8,32 +13,42 @@ import { resolveObject } from './utils/componentUtils.js';
+  'Get methods for singleton resource should return data for a single resource. This method returns an array or a paginated response. If this is not a singleton resource, please implement all standard methods.';
 
 export default (input, _, { path, documentInventory }) => {
   const oas = documentInventory.resolved;
   const resourcePath = path[1];
-  const resourcePaths = getResourcePaths(resourcePath, Object.keys(oas.paths));
 
-  if (isCustomMethod(resourcePath) || (!isChild(resourcePath) && !isSingletonResource(resourcePaths))) {
-    return;
-  }
+  const isSingleResource = isSingleResourceIdentifier(resourcePath);
+  const isSingleton =
+    isResourceCollectionIdentifier(resourcePath) && isSingletonResource(getResourcePathItems(resourcePath, oas.paths));
 
-  const errors = [];
+  if (isSingleResource || isSingleton) {
+    const errors = [];
 
-  const responseSchemas = getAllSuccessfulResponseSchemas(input);
-  responseSchemas.forEach(({ schemaPath, schema }) => {
-    const fullPath = path.concat(schemaPath);
-    const responseObject = resolveObject(oas, fullPath);
+    const responseSchemas = getAllSuccessfulResponseSchemas(input);
+    responseSchemas.forEach(({ schemaPath, schema }) => {
+      const fullPath = path.concat(schemaPath);
+      const responseObject = resolveObject(oas, fullPath);
 
-    if (hasException(responseObject, RULE_NAME)) {
-      collectException(responseObject, RULE_NAME, fullPath);
-    } else if (schemaIsPaginated(schema) || schemaIsArray(schema)) {
-      collectAndReturnViolation(fullPath, RULE_NAME, ERROR_MESSAGE_STANDARD_RESOURCE);
-      errors.push({ path: fullPath, message: ERROR_MESSAGE_STANDARD_RESOURCE });
-    } else {
-      collectAdoption(fullPath, RULE_NAME);
-    }
-  });
+      if (hasException(responseObject, RULE_NAME)) {
+        collectException(responseObject, RULE_NAME, fullPath);
+      } else if (schemaIsPaginated(schema) || schemaIsArray(schema)) {
+        collectAndReturnViolation(
```

When you `return collectAndReturnViolation`, it should return the error. Didn't it work that way? Is there a reason not returning here?

---

Okay, I see now it is inside `forEach`, ignore my comment